algorithms/a3c/test.py

```diff
@@ -0,0 +1,83 @@
+"""
+Adapted from https://github.com/ikostrikov/pytorch-a3c
+"""
+
+import time
+from collections import deque
+
+import torch
+import torch.nn.functional as F
+from torch.autograd import Variable
+
+from envs import create_atari_env
+from gym_ai2thor.envs.ai2thor_env import AI2ThorEnv
+from model import ActorCritic
+
+
+def test(rank, args, shared_model, counter):
+    torch.manual_seed(args.seed + rank)
+
+    if args.atari:
+        env = create_atari_env(args.atari_env_name)
+    else:
+        args.config_dict = {'max_episode_length': args.max_episode_length}
+        env = AI2ThorEnv(config_dict=args.config_dict)
+    env.seed(args.seed + rank)
+
+    model = ActorCritic(env.observation_space.shape[0], env.action_space.n, args.frame_dim)
+
+    model.eval()
+
+    state = env.reset()
+    state = torch.from_numpy(state)
+    reward_sum = 0
+    done = True
+
+    start_time = time.time()
+
+    # a quick hack to prevent the agent from stucking
+    actions = deque(maxlen=100)
+    episode_length = 0
+    while True:
+        episode_length += 1
+        if args.atari and args.atari_render:
+            env.render()
+        # Sync with the shared model
+        if done:
+            model.load_state_dict(shared_model.state_dict())
+            cx = Variable(torch.zeros(1, 256), volatile=True)
+            hx = Variable(torch.zeros(1, 256), volatile=True)
+        else:
+            cx = Variable(cx.data, volatile=True)
+            hx = Variable(hx.data, volatile=True)
+
+        value, logit, (hx, cx) = model((Variable(state.unsqueeze(0).float(), volatile=True),
+                                        (hx, cx)))
+        prob = F.softmax(logit)
```

Comment:
This use of softmax is deprecated, give the dim=1

Reply:
`dim=-1`